### PR TITLE
🐛 Fix new comment dropdown issue

### PR DESCRIPTION
**Pull request by Link2Twenty**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
When you create a new comment on the comments page then try and access its menu no menu opens. This was caused by an error in JS.

## Related Tickets & Documents
Loosely related to #13365

### Before
forem_new-comment-issue

### After
forem_fix-new-comment

### UI accessibility concerns?
N/A

## Added tests?

- [ ] Yes
- [x] No, and this is why: old tests will still work
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

Not a gross bug anymore

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeCommentsPage.js.erb [hunk 1]
```diff
@@ -557,8 +557,12 @@ function listenForDetailsToggle() {
 }
 
 function updateCommentsCount() {
-  var commentsCountDiv = document.getElementsByClassName("js-comments-count")[0];
-  var commentsCountData = parseInt(commentsCountDiv.dataset.commentsCount) + 1;
+  const commentsCountDiv = document.getElementsByClassName("js-comments-count")[0];
```
Comment (nickytonline) [inline app/assets/javascripts/initializers/initializeCommentsPage.js.erb:560]: Refactor (blocking): Use document.querySelector('.js-comments-count').

File: app/assets/javascripts/initializers/initializeCommentsPage.js.erb [hunk 2]
```diff
@@ -557,8 +557,12 @@ function listenForDetailsToggle() {
...
+
+  // if there's nowhere to put the count return early.
+  if(!commentsCountDiv) return;
+
+  const commentsCountData = parseInt(commentsCountDiv.dataset.commentsCount) + 1;
```
Comment (nickytonline) [inline app/assets/javascripts/initializers/initializeCommentsPage.js.erb:565]: Refactor (blocking): Add a [radix](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#parameters) to `parseInt` for base 10, e.g. `parseInt(commentsCountDiv.dataset.commentsCount, 10)` or use `Number(commentsCountDiv.dataset.commentsCount)`

image

**Review by nickytonline — CHANGES_REQUESTED** [review]
Thanks for the PR @Link2Twenty! A couple change requests.

**Comment by Link2Twenty** [comment]
For future reference, when fixing a bug like this should I modernise adjacent code, even if it's not code I'm actively working on?

**Comment by nickytonline** [comment]
> For future reference, when fixing a bug like this should I modernise adjacent code, even if it's not code I'm actively working on?

Thanks for asking @Link2Twenty! I just mentioned it as it's a small fix since we're in this piece of code. If it was a larger refactor noticed while fixing an issue, I'd create a separate issue for that.

**Review by nickytonline — APPROVED** [review]
Thanks @Link2Twenty! 🚀

**Review by ludwiczakpawel — APPROVED** [review]

**Pull request merged**